src/pool/srv_pool.c

```diff
@@ -1849,16 +1851,23 @@ pool_svc_update_map_metrics(uuid_t uuid, struct pool_map *map, struct pool_metri
 	}
 	d_tm_set_gauge(metrics->disabled_targets, num_disabled);
 
-	rc = pool_map_find_upin_tgts(map, NULL, &num_enabled);
+	rc = pool_map_find_tgts_by_state(map, PO_COMP_ST_DRAIN, NULL, &num_draining);
 	if (rc != 0) {
-		DL_ERROR(rc, DF_UUID ": failed to get upin targets", DP_UUID(uuid));
+		DL_ERROR(rc, DF_UUID ": failed to get draining targets", DP_UUID(uuid));
 		D_GOTO(out, rc);
 	}
-	d_tm_set_gauge(metrics->total_targets, num_enabled + num_disabled);
+	d_tm_set_gauge(metrics->draining_targets, num_draining);
+
+	rc = pool_map_find_tgts_by_state(map, -1, NULL, &num_total);
```

Comment:
I wonder if this could (I'm not 100% sure here without trying it directly) be replaced with a call to pool_map_target_nr(struct pool_map *map) that seems to be able to get the total number of targets in the pool map without having to loop and match on state criteria?

Reply:
Will check this and create a followup if it works.